src/utilities/findBreakingChanges.js

```diff
@@ -363,3 +545,36 @@ export function findValuesRemovedFromEnums(
   });
   return valuesRemovedFromEnums;
 }
+
+export function findInterfacesRemovedFromObjectTypes(
+  oldSchema: GraphQLSchema,
+  newSchema: GraphQLSchema
+): Array<BreakingChange> {
+  const oldTypeMap = oldSchema.getTypeMap();
+  const newTypeMap = newSchema.getTypeMap();
+  const breakingChanges = [];
+
+  Object.keys(oldTypeMap).forEach(typeName => {
+    const oldType = oldTypeMap[typeName];
+    const newType = newTypeMap[typeName];
+    if (
+      !(oldType instanceof GraphQLObjectType) ||
+      !(newType instanceof oldType.constructor)
```

Comment:
I see this condition or variants like it throughout this code and am confused as to what it's attempting to compare. IIUC, this condition applies if `oldType` is not an Object type, otherwise if it is an Object type it applies if `newType` is an instanceof the `oldType`'s constructor (which can only be `GraphQLObjectType` at that point). Could conditions like this be factored into descriptive functions?